src/core/request-body-search.js

```diff
@@ -88,6 +90,21 @@ class RequestBodySearch {
         return this;
     }
 
+    /**
+     * Sets knn on the search request body.
+     *
+     * @param {Knn|Knn[]} knn
+     * @returns {RequestBodySearch} returns `this` so that calls can be chained.
+     */
+    kNN(knn) {
```

Comment:
Changes made in this file need tests.

Reply:
Lines that need coverage in `knn.js`: https://coveralls.io/builds/66407636/source?filename=src%2Fcore%2Fknn.js